umad-09-invoice.md

```diff
@@ -0,0 +1,143 @@
+initiator of a payment. For example, merchant checkout flows, a bill from any vendor or service
+provider, etc.
+
+## Invoice Format
+
+The full structure of the invoice is:
+
+```raw
+{
+    ReceiverUma: string, 
+    // Receiving UMA address
+
+    InvoiceUUID: string,
+    // Invoice UUID Served as both the identifier of the UMA invoice, and the validation of proof of payment.
+
+    Amount: number,
+    // The amount of invoice to be paid in the smalest unit of the ReceivingCurrency.
+
+    ReceivingCurrency: {
+        Code: string, 
+        // Code is the ISO 4217 (if applicable) currency code (eg. "USD"). For cryptocurrencies, 
+        // this will be a ticker symbol, such as BTC for Bitcoin.
+
+        Name: string, 
+        // Name is the full display name of the currency (eg. US Dollars).
+
+        Symbol: string,
+        // Symbol is the symbol of the currency (eg. $ for USD).
+
+        Decimals: number,
+        // Decimals is the number of digits after the decimal point for display on the sender side
+    },
+    // The currency of the invoice
+
+    Expiration: number,
+    // The unix timestamp the UMA invoice expires
+
+    IsSubjectToTravelRule: boolean,
+    // Indicates whether the VASP is a financial institution that requires travel rule information.
+
+    RequiredPayerData: {
+        "identifier": { "mandatory": true },
+        "name": { "mandatory": boolean },
+        "email": { "mandatory": boolean },
+        // Indicates we want TR data and/or utxos. This is UMA-specific and not in LUD-18. 
+        "compliance": { "mandatory": boolean },
+    }
+    // Required data about the payer. See LUD-18 for details.
+
+    UmaVersions: string,
+    // The versions of the UMA protocol supported by the VASP. It should include the highest minor
+    // version for all major versions supported by the VASP, separated by commas.
+
+    CommentCharsAllowed: number,
+    // Optional field. CommentCharsAllowed is the number of characters that the sender can include 
+    // in the comment field of the pay request.
+
+    SenderUma: string,
+    // Optional field. The sender's UMA address. If this field presents, the UMA invoice should 
+    // directly go to the sending VASP instead of showing in other formats.
+
+    MaxNumPayments: number,
+    // Optional field. The maximum number of the invoice can be paid. If this field doesn't present,
+    // there's no limit for the invoice.
+
+    KycStatus: KycStatus, 
+    // Optional field. [enum] KYC state indicating whether the receiver is a KYC'd customer of VASP2.
+
+    Callback: string
+    // The callback url that the sender should send the PayRequest to.
+
+    Signature: bytes
+    // The signature of the UMA invoice
+}
+
+```
+
+## Proof of Payment
+
+When creating the lightning invoice from the UMA invoice, the invoice UUID should be included in the
+metadata field. It should be one extra item for the metadata field. See the LUD-18 for details.
```

Comment:
nit: LUD-18 is payerdata, but metadata going into the invoice is actually just from LUD-06